client-core/src/mnemonic.rs

```diff
@@ -38,8 +38,19 @@ impl Mnemonic {
     /// Generate and returns mnemonic words
     #[allow(clippy::new_without_default)]
     #[inline]
-    pub fn new() -> Self {
-        let mnemonic = bip39::Mnemonic::new(MNEMONIC_TYPE, MNEMONIC_LANGUAGE);
+    pub fn new(mnemonics_word_count: u32) -> Self {
+        let flag = if 12 == mnemonics_word_count {
```

Comment:
maybe pattern match can make it a bit simpler?
```
let flag = match mnemonics_word_count {
  12 => MnemonicType::Words12,
  15 => MnemonicType::Words15,
...
  _ => ...
};

Reply:
that's simpler, i'll change it